src/client/auth/x509.rs

```diff
@@ -1,14 +1,12 @@
 use crate::{
     bson::doc,
-    cmap::{Command, Connection},
+    cmap::{Command, CommandResponse, Connection},
     error::{Error, Result},
     options::Credential,
 };
 
-pub(super) async fn authenticate_stream(
-    conn: &mut Connection,
-    credential: &Credential,
-) -> Result<()> {
+/// Constructs the first client message in the X.509 handshake.
+pub(crate) fn build_client_first(credential: &Credential) -> Command {
```

Comment:
Similar to SCRAM, I made a separate method to build the first client message that can be used for speculative auth. Since we don't need to inspect the contents of this later like we need to with the first SCRAM message, I didn't bother defining a separate type this.